rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala

```diff
@@ -0,0 +1,216 @@
+package coop.rchain.rholang.interpreter
+
+import com.google.protobuf.ByteString
+import com.google.protobuf.ByteString.ByteIterator
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.models.Channel.ChannelInstance
+import coop.rchain.models.Channel.ChannelInstance.{ChanVar, Quote}
+import coop.rchain.models.Expr.ExprInstance
+import coop.rchain.models.Expr.ExprInstance._
+import coop.rchain.models.TaggedContinuation.TaggedCont.ScalaBodyRef
+import coop.rchain.models.Var.VarInstance.FreeVar
+import coop.rchain.models._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.interpreter.storage.implicits._
+import coop.rchain.rspace.{ISpace, IStore}
+import monix.eval.Task
+import scala.collection.immutable.Seq
+import scala.collection.{Seq => RootSeq}
+
+class Registry(private val space: ISpace[Channel,
```

Comment:
Re commit message:
`This is just a patricia trie with byte fanout. At the moment only lookup is
implemented.`

I really can't see the places where it's a patricia tree, maybe b/c of only lookup being implemented :) I've got to take that for a ride and try to understand/restructure the AST work in here. I think that's what's preventing me from seeing the tree structure.

Reply:
Nodes are maps.
The keys in the map are byte arrays of length [0, 1]
The values in the map are tuples:
(tag, edgeAdditional, data)
tag:
if tag is 0, it is a terminal edge
if tag is 1, data is a name where another map can be read.
edgeAdditional:
this is a bytestring that stores any additional edge labeling in the radix tree.
data:
if tag is 0, this is the stored data.
if tag is 1, this is a name where the process recurs.

Reply:
:+1: Please paste this into the `Registry`'s javadoc

Reply:
Done.